src/hooks/number-validator/number-validator.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+
+const numberRegex = RegExp(/^-?(0|[1-9]\d*)?(\.\d+)?(?<=\d)$/);
```

Comment:
I might be overthinking it but there are places in the world that use `,` as a delimiter in their numbers. Do we support that?

Examples:

```
Many European countries use a comma in place of the decimal point and use periods or blank spaces to separate every third digit.

United States: 2,367.48 francs

France: 2.367,48 francs or 2 367,48 francs
```

Reply:
This also plays into brower's locale functionality. I think `type="number"` might support locales but we do have problem with invalid input and the empty string for `onChange`. 🤔 Though, I just had a thought: do we care about `onChange` if we are using browser built-in validations for `type=number`?

Reply:
Oh, I just found something cool that we might be able to leverage: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl

I think if we leverage either `type="number"` or `window.Intl` then we might be able to accomplish a desirable user experience. We can play with both and ask design for their recommendation.

Reply:
@juanca this regex does not support the European delimiters. Is that something we need to support?

The problem with the built-in browser validations is that they're not very strict, it'd be pretty easy to have bad input that could be caught on the frontend before going to the backend at all.

I'll have a look at that link on the morrow.

Reply:
I'm not sure if we need to support it. What does the existing custom fields do? Is our bigmaven backend smart enough to parse out different formats? This might also play into it. But my intuition is to support varying formats since our application is used around the globe.

Reply:
Fair enough, seems like some good product/design questions for @annemnichols. 😉

Reply:
After talking this over with Anne and asking some questions of the fine folks over at M-Who, it seems like we could, do, and probably should support European-style delimiters. I'll put that in.

Reply:
I'm not sure I see this bits of code. Let me know where to look.